packages/legend-query/src/components/QueryBuilderPropertyExpressionEditor.tsx

```diff
@@ -114,16 +127,39 @@ const DerivedPropertyParameterValueEditor = observer(
             </div>
           )}
           <QueryBuilderValueSpecificationEditor
-            valueSpecification={parameterValue as ValueSpecification}
-            graph={
-              derivedPropertyExpressionState.queryBuilderState.graphManagerState
-                .graph
-            }
-            expectedType={
-              parameterValue?.genericType?.value.rawType ??
-              derivedPropertyExpressionState.propertyExpression.func.genericType
-                .value.rawType
+            valueSpecification={
+              derivedPropertyExpressionState.parameterValues[
+                idx
+              ] as ValueSpecification
             }
+            updateValueSpecification={(
+              val: ValueSpecification | undefined,
+            ): void => {
+              if (val) {
```

Comment:
lets not clutter the Component with this logic. Could we move this up to a declared function. Looks a bit cleaner.